content/agent/docker/autodiscovery.md

```diff
@@ -0,0 +1,268 @@
+- [Etcd](https://github.com/DataDog/integrations-core/blob/master/etcd/auto_conf.yaml)
+- [Kubernetes_state](https://github.com/DataDog/integrations-core/blob/master/kubernetes_state/auto_conf.yaml)
+- [Kube_dns](https://github.com/DataDog/integrations-core/blob/master/kube_dns/auto_conf.yaml)
+- [Kyototycoon](https://github.com/DataDog/integrations-core/blob/master/kyototycoon/auto_conf.yaml)
+- [Memcached](https://github.com/DataDog/integrations-core/blob/master/mcache/auto_conf.yaml)
+- [Redis](https://github.com/DataDog/integrations-core/blob/master/redisdb/auto_conf.yaml)
+- [Riak](https://github.com/DataDog/integrations-core/blob/master/riak/auto_conf.yaml)
+
+These templates may suit you in basic cases, but if you need to use custom check configurations—say you want to enable extra check options, use different container identifiers, or use template variable indexing— you'll have to write your own auto-conf files. You can then provide those in a few ways:
+
+1. Add them to each host that runs docker-datadog-agent and [mount the directory that contains them](https://github.com/DataDog/datadog-agent) into the docker-dd-agent container when starting it
+2. On Kubernetes, add them using [ConfigMaps](/agent/kubernetes/container_installation/#configmap )
+
+### Example: Apache check
+
+Here's the `apache.yaml` template packaged with docker-dd-agent:
+
+```yaml
+ad_identifiers:
+  - httpd
+
+init_config:
+
+instances:
+  - apache_status_url: http://%%host%%/server-status?auto
+```
+
+It looks like a minimal [Apache check configuration](https://github.com/Datadog/integrations-core/blob/master/apache/conf.yaml.example), but notice the `ad_identifiers` option. This required option lets you provide container identifiers. Autodiscovery applies this template to any containers on the same host that run an `httpd` image.
+
+_Any_ `httpd` image. Suppose you have one container running `library/httpd:latest` and another running `yourusername/httpd:v2`. Autodiscovery applies the above template to both containers. When it's loading auto-conf files, Autodiscovery cannot distinguish between identically-named images from different sources or with different tags, and **you have to provide short names for container images**, e.g. `httpd`, NOT `library/httpd:latest`.
+
+If this is too limiting—if you need to apply different check configurations to different containers running the same image— [use labels to identify the containers](/agent/kubernetes/autodiscovery/#template-source-kubernetes-pod-annotations). Label each container differently, then add each label to any template file's `ad_identifiers` list (yes, `ad_identifiers` is where to put _any_ kind of container identifier, not just images).
+
+### Template Source: Key-value Store
+
+Autodiscovery can use [Consul](/integrations/consul), etcd, and Zookeeper as template sources. To use a key-value store, you must configure it in `datadog.yaml` or in environment variables passed to the docker-dd-agent container.
```

Comment:
Any reason we link to the Consul integration, but not etcd and Zookeeper?

Reply:
ping @mfpierre

Reply:
Nope, let's add link to `/integrations/etcd` & `/integrations/zk`

⚠️ btw the `etcd` integration holds the `ceph` description had a hard time finding it on the page